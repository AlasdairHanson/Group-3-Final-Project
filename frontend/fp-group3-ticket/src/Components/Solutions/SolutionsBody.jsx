import React from "react";
const SolutionsBody = ({ content }) => {
  return (
    <>
      <p>{content}</p>
    </>
  );
};
export default SolutionsBody;
