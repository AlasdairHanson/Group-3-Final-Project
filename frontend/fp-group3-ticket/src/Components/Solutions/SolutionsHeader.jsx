import React from "react";
const SolutionsHeader = ({ title }) => {
  return <h2 className="solutionsHeader">{title}</h2>;
};
export default SolutionsHeader;
