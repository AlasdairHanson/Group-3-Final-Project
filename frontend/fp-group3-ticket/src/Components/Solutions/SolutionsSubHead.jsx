import React from "react";
const SolutionsSubHead = ({ title }) => {
  return (
    <>
      <h3>{title}</h3>
    </>
  );
};
export default SolutionsSubHead;
