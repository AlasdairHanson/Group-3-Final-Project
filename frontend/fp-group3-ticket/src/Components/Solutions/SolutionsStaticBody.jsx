import React from "react";
import SolutionsBody from "./SolutionsBody";
import SolutionsHeader from "./SolutionsHeader";
import SolutionsSubHead from "./SolutionsSubHead";
const SolutionsStaticBody = () => {
  return (
    <>
      <div className="solutionsBody">
        <SolutionsHeader title="Ut posuere eu neque vel blandit. Nulla. " />
        <SolutionsSubHead title="Mauris pretium hendrerit lectus molestie aliquam. Sed. " />
        <SolutionsBody content="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod convallis metus ultricies tincidunt. Suspendisse potenti. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Ut maximus sem vel sapien aliquet, quis rhoncus arcu eleifend. Quisque in diam nunc. Phasellus et fringilla mi. Curabitur aliquet neque auctor accumsan sollicitudin. Vestibulum pellentesque ut dui sit amet mollis. Nulla at efficitur dolor, sit amet convallis nisl. Nulla auctor pulvinar nibh quis tempor. Donec auctor orci arcu, eget accumsan metus iaculis pulvinar. Duis porta sagittis dolor, nec vulputate urna. Nam venenatis consectetur lectus ut rhoncus. Phasellus ornare nunc eu luctus ultrices. Sed vitae dapibus neque. Maecenas dui tellus, cursus a dui ac, faucibus finibus dolor. Mauris eu turpis neque. Duis nulla nibh, egestas vel fringilla et, elementum non risus. " />
        <SolutionsHeader title="Ut posuere eu neque vel blandit. Nulla. " />
        <SolutionsSubHead title="Mauris pretium hendrerit lectus molestie aliquam. Sed. " />
        <SolutionsBody content="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod convallis metus ultricies tincidunt. Suspendisse potenti. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Ut maximus sem vel sapien aliquet, quis rhoncus arcu eleifend. Quisque in diam nunc. Phasellus et fringilla mi. Curabitur aliquet neque auctor accumsan sollicitudin. Vestibulum pellentesque ut dui sit amet mollis. Nulla at efficitur dolor, sit amet convallis nisl. Nulla auctor pulvinar nibh quis tempor. Donec auctor orci arcu, eget accumsan metus iaculis pulvinar. Duis porta sagittis dolor, nec vulputate urna. Nam venenatis consectetur lectus ut rhoncus. Phasellus ornare nunc eu luctus ultrices. Sed vitae dapibus neque. Maecenas dui tellus, cursus a dui ac, faucibus finibus dolor. Mauris eu turpis neque. Duis nulla nibh, egestas vel fringilla et, elementum non risus. " />
        <SolutionsHeader title="Ut posuere eu neque vel blandit. Nulla. " />
        <SolutionsSubHead title="Mauris pretium hendrerit lectus molestie aliquam. Sed. " />
        <SolutionsBody content="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean euismod convallis metus ultricies tincidunt. Suspendisse potenti. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Ut maximus sem vel sapien aliquet, quis rhoncus arcu eleifend. Quisque in diam nunc. Phasellus et fringilla mi. Curabitur aliquet neque auctor accumsan sollicitudin. Vestibulum pellentesque ut dui sit amet mollis. Nulla at efficitur dolor, sit amet convallis nisl. Nulla auctor pulvinar nibh quis tempor. Donec auctor orci arcu, eget accumsan metus iaculis pulvinar. Duis porta sagittis dolor, nec vulputate urna. Nam venenatis consectetur lectus ut rhoncus. Phasellus ornare nunc eu luctus ultrices. Sed vitae dapibus neque. Maecenas dui tellus, cursus a dui ac, faucibus finibus dolor. Mauris eu turpis neque. Duis nulla nibh, egestas vel fringilla et, elementum non risus. " />
      </div>
    </>
  );
};
export default SolutionsStaticBody;
