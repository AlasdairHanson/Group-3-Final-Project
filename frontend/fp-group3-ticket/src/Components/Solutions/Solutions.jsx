import React from "react";
import { Col, Container, Row } from "react-bootstrap";
import Sidebar from "../SideBar/Sidebar";
import SolutionsStaticBody from "./SolutionsStaticBody";
const Solutions = () => {
  return (
    <>
      <Container fluid={true}>
        <Row className="noMargin">
          <Col xs={2} className="mt-3 pt-3 pl-0">
            <Sidebar />
          </Col>
          <Col xs={10} className="pt-3 mt-3 pr-0">
            <SolutionsStaticBody />
          </Col>
        </Row>
      </Container>
    </>
  );
};
export default Solutions;
