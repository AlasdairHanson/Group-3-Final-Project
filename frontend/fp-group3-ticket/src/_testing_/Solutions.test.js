import { render, screen } from '@testing-library/react';
import Solutions from '../Components/Solutions/Solutions';
import '@testing-library/jest-dom';
import {create} from 'react-test-renderer';

jest.mock(`../Components/Solutions/SolutionsStaticBody`, () => {
  return function SolutionsStaticBodyMock(props){
    return(<p>Mocked SolutionsStaticBody</p>)
  }
});

describe(`Snapshot test of Solutions`, () => {

  test(` Solutions should match the snapshot every time it is rendered`, () => {
    const instance = create(<Solutions/>).toJSON(); 
    expect(instance).toMatchSnapshot();
  });
});