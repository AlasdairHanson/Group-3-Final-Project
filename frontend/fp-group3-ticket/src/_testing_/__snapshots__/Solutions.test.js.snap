// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of Solutions  Solutions should match the snapshot every time it is rendered 1`] = `
<div
  className="container-fluid"
>
  <div
    className="noMargin row"
  >
    <div
      className="mt-3 pt-3 pl-0 col-2"
    >
      <div
        className="parentSideContainer"
      >
        <div
          className="sidebarContainer mb-3"
        >
          <div
            className="accordion"
          >
            <div
              className="sideBarContainer card"
            >
              <div
                className="p-0 sideBarContainer card-header"
                onClick={[Function]}
              >
                <div
                  className="sidebarInnerContainer"
                >
                  <h4
                    className="sidebarText"
                  >
                    Sort
                  </h4>
                </div>
              </div>
              <div
                aria-expanded={null}
                className="collapse"
              >
                <div
                  className="p-0 card-body"
                >
                  <div
                    className="sidebarInnerContainer sideBody"
                    onClick={[Function]}
                  >
                    <p
                      className="sidebarChildText"
                    >
                      Title
                    </p>
                  </div>
                  <div
                    className="sidebarInnerContainer sideBody"
                    onClick={[Function]}
                  >
                    <p
                      className="sidebarChildText"
                    >
                      Topic
                    </p>
                  </div>
                  <div
                    className="sidebarInnerContainer sideBody"
                    onClick={[Function]}
                  >
                    <p
                      className="sidebarChildText"
                    >
                      Cohort
                    </p>
                  </div>
                  <div
                    className="sidebarInnerContainer sideBody"
                    onClick={[Function]}
                  >
                    <p
                      className="sidebarChildText"
                    >
                      Author
                    </p>
                  </div>
                  <div
                    className="sidebarInnerContainer sideBody"
                    onClick={[Function]}
                  >
                    <p
                      className="sidebarChildText"
                    >
                      Trainer
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sidebarContainer mb-3"
        >
          <div
            className="accordion"
          >
            <div
              className="sideBarContainer card"
            >
              <div
                className="p-0 sideBarContainer card-header"
                onClick={[Function]}
              >
                <div
                  className="sidebarInnerContainer"
                >
                  <h4
                    className="sidebarText"
                  >
                    Filter
                  </h4>
                </div>
              </div>
              <div
                aria-expanded={null}
                className="collapse"
              >
                <div
                  className="p-0 card-body"
                >
                  <div
                    className="accordion"
                  >
                    <div
                      className="sideBarContainer card"
                    >
                      <div
                        className="p-0 sideBarContainer card-header"
                        onClick={[Function]}
                      >
                        <div
                          className="sidebarInnerContainer"
                        >
                          <h6
                            className="sidebarText"
                          >
                            Priority
                          </h6>
                        </div>
                      </div>
                      <div
                        aria-expanded={null}
                        className="collapse"
                      >
                        <div
                          className="p-0 card-body"
                        >
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Low
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Medium
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              High
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Critical
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="accordion"
                  >
                    <div
                      className="sideBarContainer card"
                    >
                      <div
                        className="p-0 sideBarContainer card-header"
                        onClick={[Function]}
                      >
                        <div
                          className="sidebarInnerContainer"
                        >
                          <h6
                            className="sidebarText"
                          >
                            Status
                          </h6>
                        </div>
                      </div>
                      <div
                        aria-expanded={null}
                        className="collapse"
                      >
                        <div
                          className="p-0 card-body"
                        >
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Unresolved
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Ongoing
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Resolved
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="accordion"
                  >
                    <div
                      className="sideBarContainer card"
                    >
                      <div
                        className="p-0 sideBarContainer card-header"
                        onClick={[Function]}
                      >
                        <div
                          className="sidebarInnerContainer"
                        >
                          <h6
                            className="sidebarText"
                          >
                            Topic
                          </h6>
                        </div>
                      </div>
                      <div
                        aria-expanded={null}
                        className="collapse"
                      >
                        <div
                          className="p-0 card-body"
                        >
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Networking
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Security
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Privacy
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Hardware
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Other
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="accordion"
                  >
                    <div
                      className="sideBarContainer card"
                    >
                      <div
                        className="p-0 sideBarContainer card-header"
                        onClick={[Function]}
                      >
                        <div
                          className="sidebarInnerContainer"
                        >
                          <h6
                            className="sidebarText"
                          >
                            Cohort
                          </h6>
                        </div>
                      </div>
                      <div
                        aria-expanded={null}
                        className="collapse"
                      >
                        <div
                          className="p-0 card-body"
                        >
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              CloudNative
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Software
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              DevOps
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Testing
                            </p>
                          </div>
                          <div
                            className="sidebarInnerContainer sideBody"
                            onClick={[Function]}
                          >
                            <p
                              className="sidebarChildText"
                            >
                              Other
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pt-3 mt-3 pr-0 col-10"
    >
      <p>
        Mocked SolutionsStaticBody
      </p>
    </div>
  </div>
</div>
`;
